packages/experimental/Menu/src/MenuList/MenuList.tsx

```diff
@@ -33,18 +35,53 @@ export const MenuList = compose<MenuListType>({
   ...stylingSettings,
   slots: {
     root: MenuStack,
+    focusZone: FocusZone,
   },
   useRender: (userProps: MenuListProps, useSlots: UseSlots<MenuListType>) => {
     const menuList = useMenuList(userProps);
     const contextValue = useMenuListContextValue(menuList);
     const Slots = useSlots(menuList);
 
+    /**
+     * On macOS, focus isn't placed by default on the first focusable element. We get around this by focusing on the inner FocusZone
+     * hosting the menu. For whatever reason, to get the timing _just_ right to actually focus, we need an additional `setTimeout`
+     *  on top of the `useLayoutEffect` hook.
+     */
+    const focusZoneRef = React.useRef<IFocusable>(null);
```

Comment:
I think it would be better to group all the hook code under useMenuList, so that it's all in one place to find.

Reply:
Right. I think I got tripped up copying code from `usePrepareProps` in the old compose framework to here. I'll update accordingly. I think the `useLayoutEffect` call might have to stay in the render of MenuList though?

Reply:
Why's that? Is there a rule that it has to be in a tsx file?

Reply:
I was wrong. I thought it had to be in render but I guess you can call hooks from hooks.